gen/schema_gen.go

```diff
@@ -266,6 +270,28 @@ func (g *schemaGen) generate2(name string, schema *jsonschema.Schema) (ret *ir.T
 		}
 	}
 
+	if schema.XOgenType != "" {
+		t, err := ir.External(schema)
+		if err != nil {
+			return nil, errors.Wrap(err, "external type")
+		}
+
+		if pkg, ok := g.imports[path.Base(t.External.PackagePath)]; ok && pkg != t.External.PackagePath {
+			for i := 0; true; i++ {
+				importAlias := fmt.Sprintf("%s%d", path.Base(t.External.PackagePath), i)
+				if pkg, ok := g.imports[importAlias]; !ok || pkg == t.External.PackagePath {
+					t.External.ImportAlias = importAlias
+					t.Primitive = ir.PrimitiveType(importAlias + "." + t.External.GoName)
+					break
+				}
+			}
+		}
+
+		g.imports[cmp.Or(t.External.ImportAlias, path.Base(t.External.PackagePath))] = t.External.PackagePath
```

Comment:
```suggestion
                pkgPath := t.External.PackagePath
                pkgName := path.Base(pkgPath)
                
		if pkg, ok := g.imports[pkgName]; ok && pkg != pkgPath {
			for i := 0; true; i++ {
				importAlias := fmt.Sprintf("%s%d", pkgName, i)
				if pkg, ok := g.imports[importAlias]; !ok || pkg == pkgPath {
					t.External.ImportAlias = importAlias
					t.Primitive = ir.PrimitiveType(importAlias + "." + t.External.GoName)
					break
				}
			}
		}

		g.imports[cmp.Or(t.External.ImportAlias, pkgName)] = pkgPath
```

Reply:
Have amended although it's slightly different now as I added support for builtin types.